relay-config/src/config.rs

```diff
@@ -169,7 +169,18 @@ trait ConfigObject: DeserializeOwned + Serialize {
 
     /// The full filename of the config file, including the file extension.
     fn path(base: &Path) -> PathBuf {
-        base.join(format!("{}.{}", Self::name(), Self::format().extension()))
+        let extensions = Self::format().extensions();
+
+        for extension in &extensions[..extensions.len() - 1] {
```

Comment:
```suggestion
        for extension in extensions {
```

Reply:
But in this case the last element couldn't be checked twice?

Reply:
You're right, there's no need to check the last element since the function falls back to it.